utils/llm.py

```diff
@@ -147,23 +136,26 @@ def make_assistant_message(content: str) -> LLMMessage:
 
 
 def context_token_len(context: LLMContext, model: Optional[str] = None, response: bool = False) -> int:
```

Comment:
It looks like [`litellm.token_counter`](https://litellm.vercel.app/docs/completion/token_usage#3-token_counter) takes care of this.